From 374b09cb60d46f1ecb784ef6d7a8e506bc0b9552 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Thu, 7 Nov 2024 20:57:00 +0100
Subject: [PATCH] patch

---
 speech_recognition/audio.py | 34 ++++------------------------------
 1 file changed, 4 insertions(+), 30 deletions(-)

diff --git a/speech_recognition/audio.py b/speech_recognition/audio.py
index 2322f36..8a36559 100644
--- a/speech_recognition/audio.py
+++ b/speech_recognition/audio.py
@@ -262,36 +262,10 @@ def get_flac_converter():
     """Returns the absolute path of a FLAC converter executable, or raises an OSError if none can be found."""
     flac_converter = shutil_which("flac")  # check for installed version first
     if flac_converter is None:  # flac utility is not installed
-        base_path = os.path.dirname(
-            os.path.abspath(__file__)
-        )  # directory of the current module file, where all the FLAC bundled binaries are stored
-        system, machine = platform.system(), platform.machine()
-        if system == "Windows" and machine in {
-            "i686",
-            "i786",
-            "x86",
-            "x86_64",
-            "AMD64",
-        }:
-            flac_converter = os.path.join(base_path, "flac-win32.exe")
-        elif system == "Darwin" and machine in {
-            "i686",
-            "i786",
-            "x86",
-            "x86_64",
-            "AMD64",
-            "arm64",
-        }:
-            flac_converter = os.path.join(base_path, "flac-mac")
-        elif system == "Linux" and machine in {"i686", "i786", "x86"}:
-            flac_converter = os.path.join(base_path, "flac-linux-x86")
-        elif system == "Linux" and machine in {"x86_64", "AMD64"}:
-            flac_converter = os.path.join(base_path, "flac-linux-x86_64")
-        else:  # no FLAC converter available
-            raise OSError(
-                "FLAC conversion utility not available - consider installing the FLAC command line application by running `apt-get install flac` or your operating system's equivalent"
-            )
-
+        raise OSError(
+            "FLAC conversion utility not available - consider installing the FLAC command line application by running `apt-get install flac` or your operating system's equivalent"
+        )
+ 
     # mark FLAC converter as executable if possible
     try:
         # handle known issue when running on docker:
